CONTRIBUTING.md

```diff
@@ -0,0 +1,158 @@
+    Found 1 extensions: ['azure-cli-iot-ext']
+    ```
+
+Please use `az --debug` if you run into any issues, or file an issue in this GitHub repo.
+
+Please refer to the [Azure CLI Extension Guide](https://github.com/Azure/azure-cli/tree/master/doc/extensions) for further information and help developing extensions.
+
+### Running Tests
+
+1. Install Dependencies
+
+    This project utilizes the following: [pytest](https://docs.pytest.org/en/latest/), [unittest](https://docs.python.org/3.6/library/unittest.html), `pytest-mock`, and `pytest-cov`.
+
+    Run the following to install them:
+
+    `pip install -r dev_requirements`
+
+1. Activate Virtual Environment
+
+    Ensure that the virtual environment you created while setting up your machine for general Azure CLI development is activated and the dev_setup.py script has been run.
+
+#### Unit Tests
+
+_Hub:_  
+`pytest azext_iot/tests/test_iot_ext_unit.py`
+
+_DPS:_  
+`pytest azext_iot/tests/test_iot_dps_unit.py`
+
+#### Integration Tests
+
+Integration tests are run against Azure resources and depend on environment variables.
+
+##### Azure Resource Setup
+
+1. Create IoT Hub
+> IMPORTANT: Your IoT Hub must be created specifically for integration tests and must not contain any devices when the tests are run.
+1. Create Files Storage - In IoT Hub, click Files, create a new Storage Account and link to an empty Container.
+1. Create IoT Hub Device Provisioning Service (DPS)
+1. Link IoT Hub to DPS - From DPS, click "Linked IoT Hub" and link the IoT Hub you just created.
+
+##### Environment Variables
+You can either manually set the environment variables or use the `pytest.ini.example` file in the root of the extension repo. To use that file, rename it to `pytest.ini`, open it and set the variables as indicated below.
+
+```
+    AZURE_TEST_RUN_LIVE=True
+    azext_iot_testrg="Resource Group that contains your IoT Hub"
+    azext_iot_testhub="IoT Hub Name"
+    azext_iot_testhub_cs="IoT Hub Connection String"
+    azext_iot_testdps="IoT Hub DPS Name"
+    azext_iot_teststorageuri="Blob Container SAS Uri"
+```
+
+`azext_iot_teststorageuri` is optional and only required when you want to test device export and file upload functionality. You can generate a SAS Uri for your Blob container using the [Azure Storage Explorer](https://azure.microsoft.com/en-us/features/storage-explorer/).  You must also configure your IoT Hub's File Upload storage container via the Azure Portal for this test to pass.
+
+
+##### IoT Hub
+
+Execute the following command to run the IoT Hub integration tests:
+
+`pytest azext_iot/tests/test_iot_ext_int.py`
+
+
+##### Device Provisioning Service
+
+Execute the following command to run the IoT Hub DPS integration tests:
+
+`pytest azext_iot/tests/test_iot_dps_int.py`
+
+#### Unit and Integration Tests Single Command
+
+Execute the following command from the root of your extension to run both Unit and Integration tests and output a Code Coverage report to a `.coveragerc` file.
```

Comment:
...and output a Code Coverage report (both to console and to a `.coverage` file) using the `.coveragerc` file for configuration.